public/app/features/explore/QueryLibrary/QueryTemplatesList.tsx

```diff
@@ -34,20 +121,52 @@ export function QueryTemplatesList() {
     );
   }
 
-  const queryTemplateRows: QueryTemplateRow[] = data.map((queryTemplate: QueryTemplate, index: number) => {
-    const datasourceRef = queryTemplate.targets[0]?.datasource;
-    const datasourceType = getDatasourceSrv().getInstanceSettings(datasourceRef)?.meta.name || '';
-    return {
-      index: index.toString(),
-      uid: queryTemplate.uid,
-      datasourceRef,
-      datasourceType,
-      createdAtTimestamp: queryTemplate?.createdAtTimestamp || 0,
-      query: queryTemplate.targets[0],
-      description: queryTemplate.title,
-      user: queryTemplate.user,
-    };
-  });
-
-  return <QueryTemplatesTable queryTemplateRows={queryTemplateRows} />;
+  return (
+    <>
+      <div className={styles.selectors}>
+        <FilterInput
+          className={styles.searchInput}
+          placeholder={t('query-library.search', 'Search by data source, query content or description')}
+          aria-label={t('query-library.search', 'Search by data source, query content or description')}
+          value={searchQuery}
+          onChange={(query) => setSearchQuery(query)}
+          escapeRegex={false}
+        />
+        <InlineLabel className={styles.label} width="auto">
+          <Trans i18nKey="query-library.datasource-names">Datasource name(s):</Trans>
+        </InlineLabel>
+        <MultiSelect
+          className={styles.multiSelect}
+          onChange={(v) => {
+            setDatasourceFilters(v);
+          }}
```

Comment:
```suggestion
          onChange={setDatasourceFilters}
```
Nit: An extra function wrapper shouldn't be necessary here.

Reply:
Ah yes the signatures are matching